wlauto/common/android/device.py

```diff
@@ -670,6 +670,18 @@ def get_device_model(self):
         except KeyError:
             return None
 
+    def is_network_connected(self, network_host='www.google.com'):
+        self.logger.debug('Checking for internet connectivity.')
+        output = adb_shell(self.adb_name, 'ping -q -w 1 -c 1 {}'.format(network_host),
```

Comment:
Changed to use self.execute instead which has a self._check_ready() call rather than adb_shell().